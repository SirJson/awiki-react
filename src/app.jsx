import React from 'react';

import WikiEditor from './components/WikiEditor';
import Sidebar from './components/Sidebar';

const DEFAULT_FILE = 'Home';

export default class App extends React.Component {
    constructor() {
        super();
        this.state = {
            currentFile: '/home/macco/mega/awiki/Home/Bücher/Bergauf_mit_Rückenwind',
        };

        this.onClickNode = this.onClickNode.bind(this);
    }

    onClickNode(relativePath) {
        console.log('onClickNode');
        console.log(relativePath);
        this.setState({
            currentFile: relativePath,
        });
        console.log(this.state);
    }

    render() {
        return (
            <div
                style={{ display: 'flex' }}
                className="app-container"
                data-tid="container"
            >
                <Sidebar onClickNode={this.onClickNode} />
                <main style={{ padding: 0, overflowY: 'hidden' }}>
                    <WikiEditor currentFile={this.state.currentFile} />
                </main>
            </div>
        );
    }
}
