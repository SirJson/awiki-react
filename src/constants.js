import os from 'os'
import path from 'path';

export const DEFAULT_PAGE = 'Home';
export const APP_NAME = 'awiki';
export const BASEPATH = path.join(os.homedir(), APP_NAME);
export const HOMEPATH = path.join(BASEPATH, DEFAULT_PAGE);
export const TRASHPATH = path.join(BASEPATH, 'Trash');
export const EXT = '.markdown';
