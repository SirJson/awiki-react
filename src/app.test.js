import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { Sidebar, NewFileDialog, OpenPageBar } from './app';

import { HOMEPATH, DEFAULT_PAGE } from './constants';

describe('OpenPageBar', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <OpenPageBar
                isOpen={false}
                setCurrentFile={HOMEPATH}
                setOpenPageBar={() => {}}
            />,
            div,
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('NewFileDialog', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <NewFileDialog
                isOpen
                onClose={() => {}}
                newFileDir=""
                setCurrentFile={() => {}}
                setUpdateSidebar={() => {}}
            />,
            div,
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('Search', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Sidebar
                currentFile={HOMEPATH}
                updateSidebar={false}
                setCurrentFile={() => {}}
                toggleNewFileDialog={() => {}}
                setNewFileDir={() => {}}
                setUpdateSidebar={() => {}}
            />, div,
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Sidebar
                currentFile={HOMEPATH}
                updateSidebar={false}
                setCurrentFile={() => {}}
                toggleNewFileDialog={() => {}}
                setNewFileDir={() => {}}
                setUpdateSidebar={() => {}}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
