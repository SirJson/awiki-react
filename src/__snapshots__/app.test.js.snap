// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<div
  className="sidebar"
>
  <div
    className="bp3-tree"
  >
    <ul
      className="bp3-tree-node-list bp3-tree-root"
    />
  </div>
</div>
`;
